exonum-java-binding/core/src/test/java/com/exonum/binding/core/storage/indices/BaseProofMapIndexProxyIntegrationTest.java

```diff
@@ -0,0 +1,806 @@
+   */
+  private static String filterBitPrefix(String prefix) {
+    String filtered = prefix.replaceAll("[ _|]", "");
+    // Check that the string is correct
+    assert filtered.matches("[01]*");
+    assert filtered.length() <= PROOF_MAP_KEY_SIZE;
+    return filtered;
+  }
+
+  /**
+   * Creates a 32-byte key from the bit prefix.
+   */
+  private static byte[] keyFromString(String prefix) {
+    BitSet keyPrefixBits = new BitSet(prefix.length());
+    for (int i = 0; i < prefix.length(); i++) {
+      char bit = prefix.charAt(i);
+      if (bit == '1') {
+        keyPrefixBits.set(i);
+      }
+    }
+    return createPrefixed(keyPrefixBits.toByteArray(), PROOF_MAP_KEY_SIZE);
+  }
+
+  /**
+   * Create a proof key of length 32 with the specified suffix.
+   *
+   * @param suffix a key suffix. Must be shorter than or equal to 32 bytes in UTF-8.
+   * @return a key, starting with zeroes and followed by the specified suffix encoded in UTF-8
+   */
+  private static HashCode createProofKey(String suffix) {
+    byte[] suffixBytes = bytes(suffix);
+    return createProofKey(suffixBytes);
+  }
+
+  private static HashCode createProofKey(byte... suffixBytes) {
+    byte[] proofKey = createRawProofKey(suffixBytes);
+    return HashCode.fromBytes(proofKey);
+  }
+
+  private static byte[] createRawProofKey(byte... suffixBytes) {
+    checkArgument(suffixBytes.length <= PROOF_MAP_KEY_SIZE);
+    byte[] proofKey = new byte[PROOF_MAP_KEY_SIZE];
+    System.arraycopy(suffixBytes, 0, proofKey, PROOF_MAP_KEY_SIZE - suffixBytes.length,
+        suffixBytes.length);
+    return checkProofKey(proofKey);
+  }
+
+  void runTestWithView(Function<Cleaner, View> viewFactory,
+      Consumer<ProofMapIndexProxy<HashCode, String>> mapTest) {
+    runTestWithView(viewFactory, (ignoredView, map) -> mapTest.accept(map));
+  }
+
+  private void runTestWithView(
+      Function<Cleaner, View> viewFactory,
+      BiConsumer<View, ProofMapIndexProxy<HashCode, String>> mapTest) {
+    try (Cleaner cleaner = new Cleaner()) {
+      View view = viewFactory.apply(cleaner);
+      ProofMapIndexProxy<HashCode, String> map = this.create(MAP_NAME, view);
+
+      mapTest.accept(view, map);
+    } catch (CloseFailuresException e) {
+      throw new AssertionError("Unexpected exception", e);
+    }
+  }
+
+  @Override
+  StorageIndex createOfOtherType(String name, View view) {
+    return ListIndexProxy.newInstance(name, view, StandardSerializers.string());
+  }
+
+  @Override
+  Object getAnyElement(ProofMapIndexProxy<HashCode, String> index) {
+    return index.get(key1);
+  }
+
+  @Override
+  void update(ProofMapIndexProxy<HashCode, String> index) {
+    index.put(key1, V1);
+  }
+
+  /**
+   * Creates `numOfEntries` map entries, sorted by key:
+   * [(00…0PK1, V1), (00…0PK2, V2), … (00…0PKi, Vi)].
+   */
+  List<MapEntry<HashCode, String>> createSortedMapEntries() {
+    // Use PROOF_KEYS which are already sorted.
```

Comment:
The implementation and documentation are not correct. Nowhere the base class says that getProofKeys are expected to be sorted
     and how. The usages of this method must be reviewed to determine if they need sorting.
     This method/getProofKeys and documentation must be updated accordingly.